benchmark/result/metadata_class.py

```diff
@@ -0,0 +1,41 @@
+"""This module contains the metadata of a benchmark result."""
+
+from datetime import datetime
+
+import git
+from attrs import define, field
+from attrs.validators import instance_of
+
+from baybe import __version__ as baybe_version
+from baybe.serialization.mixin import SerialMixin
+
+
+@define(frozen=True)
+class ResultMetadata(SerialMixin):
+    """The metadata of a benchmark result."""
+
+    benchmark_name: str = field(validator=instance_of(str))
+    """The name of the benchmark."""
+
+    execution_time_sec: float = field(validator=instance_of(float))
+    """The execution time of the benchmark in seconds."""
+
+    gpu_used: bool = field(validator=instance_of(bool))
+    """Whether the benchmark used a GPU."""
+
+    start_datetime: str = field(
```

Comment:
like for the uuid where I remove the factory, I think this field should not have a default since the startime *must* be set from the outside. Or to put it differently: can you tell me one reasonable situation where the factory would make sense? If not, then it must go, because it otherwise only opens the door for bugs, i.e. when the caller forgets to set its datetime